### DISCUSSION TITLE
Firebase App Distribution skips iOs distribution, although the build exists

### DISCUSSION THREAD

**Discussion by RaphaelAslanian** [codemagic-ci-cd/codemagic-docs#2013] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. I am using Workflow Editor to configure Codemagic CI/CD.
2. I am building for Android and iOs.
3. I have configured distribution for Firebase App Distribution.
4. I launch the build.

### Expected results

Proceeding to the distribution step, both Android and iOs build should end up being distributed onto Firebase App Distribution.

### Actual results

The Android app bundle is distributed onto Firebase App Distribution.

On the other hand, the iOs build is *not distributed*, but skipped, as if the build was not found:
```bash
Skip publishing Ios to Firebase App Distribution: no matching artefacts found.
Skip publishing to Firebase App Distribution: no matching artefacts found.
```
However, the Build step is successful for iOs and Android, and when connecting through SSH, I can confirm there is a Runner.app.zip file in the output folder.

➡️ I don't understand why the distribution is skipped. Can you help me with that please ? 🙏 

### Build id (optional)

64f441145c318d3ddeaaac12

**Comment by dtrdic** [comment, marked as answer]
hey @RaphaelAslanian, Firebase App Distribution does not accept zip files; it only accepts actual binaries like .ipa files. To generate the .ipa file, you will need to code-sign your app. You can find detailed instructions on how to code-sign your app in this [guide](https://docs.codemagic.io/flutter-code-signing/ios-code-signing/).

**Reply by RaphaelAslanian** [reply]
Hi @dtrdic ! Thank you, I will try that :) !

**Reply by RaphaelAslanian** [reply]
Works fine 👍 thank you !
